### PR TITLE
[moe] support low level zero optim

### PR DESCRIPTION
**Pull request by oahzxl**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [ ] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.
moe param should not be stored in working_groups because they have different parallel strategy. so we need to store them separately in param_groups instead of working_groups

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [ ] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [ ] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/zero/low_level/low_level_optim.py [hunk 1]
```diff
@@ -418,6 +434,11 @@ def step(self, closure=None):
         # update the parameters
         self.optim.step()
 
+        # release the moe grad
+        if len(self.param_groups) > len(self._working_param_groups):
+            for param in self.param_groups[-1]['params']:
+                param.grad = None
+
         # release the grad
```
Comment (ver217) [inline colossalai/zero/low_level/low_level_optim.py:442]: Can you integrate moe grad handle into this optimizer? As we won't have any grad handler anymore.

Comment (oahzxl) [reply colossalai/zero/low_level/low_level_optim.py:442]: this grad handle is only for pytorch model test. will not be used in our model

**Review by ver217 — CHANGES_REQUESTED** [review]
By the way, can you create a moe param interface like dtensor? https://github.com/hpcaitech/ColossalAI/blob/main/colossalai/tensor/d_tensor/api.py

File: colossalai/zero/low_level/low_level_optim.py (hunk 1 above)
Comment (ver217) [reply colossalai/zero/low_level/low_level_optim.py:442]: Then can we move this to `tests/`? Engine will be deprecated soon.

Comment (oahzxl) [reply colossalai/zero/low_level/low_level_optim.py:442]: ok

**Comment by oahzxl** [comment]
> By the way, can you create a moe param interface like dtensor? https://github.com/hpcaitech/ColossalAI/blob/main/colossalai/tensor/d_tensor/api.py

done

**Review by ver217 — APPROVED** [review]

**Pull request merged**